src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -114,6 +115,11 @@ private string GetRootPathWithGuid_NoLock()
                                                                      .GetDocument(temporaryProjectInfoAndDocumentId.Item2);
 
                     var useDecompiler = allowDecompilation;
+                    if (useDecompiler)
+                    {
+                        useDecompiler = !symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass.Name == nameof(SuppressIldasmAttribute));
```

Comment:
Should this be checking fully qualified name?

Reply:
My original approach was to check the fully-qualified name. The more I though about it, it didn't seem to add anything.